spec/firebase_app_distribution_auth_client_spec.rb

```diff
@@ -29,87 +29,191 @@
       .and_return(fake_binary_contents)
     allow(fake_binary_contents).to receive(:key)
       .and_return("fake_service_key")
-    allow(File).to receive(:exist?).and_return(false)
-
-    allow(ENV).to receive(:[])
-      .with("GOOGLE_APPLICATION_CREDENTIALS")
-      .and_return(nil)
-    allow(ENV).to receive(:[])
-      .with("FIREBASE_TOKEN")
-      .and_return(nil)
-    allow(ENV).to receive(:[])
-      .with("XDG_CONFIG_HOME")
-      .and_return(nil)
   end
 
   describe '#fetch_auth_token' do
-    it 'uses service credentials for authorization when the path is passed in' do
-      expect(auth_helper.fetch_auth_token("google_service_path")).to eq("service_fake_auth_token")
-    end
+    describe 'with all other auth variables as nil' do
+      before(:each) do
+        allow(ENV).to receive(:[])
+          .with("GOOGLE_APPLICATION_CREDENTIALS")
+          .and_return(nil)
+        allow(ENV).to receive(:[])
+          .with("FIREBASE_TOKEN")
+          .and_return(nil)
+        allow(ENV).to receive(:[])
+          .with("XDG_CONFIG_HOME")
+          .and_return(nil)
+      end
```

Comment:
Right now specs are doubled because we are testing nil and empty string inputs. Is there a way to make a helper method that can set these allow() calls before the specs run? Having this before(:each) block works but I'm forced to double the specs in order to set all the variables to "" for the empty string tests. It would be easier to have a helper method to set all the environment variables to nil or empty strings and call that within each spec but I am not sure if it's possible to do within a spec file